lib/python-sdk/common_grants_sdk/schemas/models/opp_transformer.py

```diff
@@ -0,0 +1,98 @@
+from abc import ABC, abstractmethod
+from datetime import datetime, UTC
+from typing import Annotated, Any, final
+from uuid import UUID, uuid4
+
+from pydantic import Field
+
+from common_grants_sdk.schemas.base import CommonGrantsBaseModel
+from common_grants_sdk.schemas.models.opp_base import OpportunityBase
+from common_grants_sdk.schemas.models.opp_funding import OppFunding
+from common_grants_sdk.schemas.models.opp_status import OppStatus
+from common_grants_sdk.schemas.models.opp_timeline import OppTimeline
+
+
+class OpportunityTransformer(CommonGrantsBaseModel, ABC):
+    """
+    Base class for transforming arbitrary data structures
+    into canonical models.
+    """
+
+    source_data: Annotated[
+        dict[str, Any],
+        Field(
+            description="Arbitrary source data",
+        ),
+    ]
+
+    def transform_opportunity(self, id: UUID | None = None) -> OpportunityBase:
+        """
+        Extract and transform opportunity data from source_data
+        into a canonical OpportunityBase model instance.
+        """
+
+        return OpportunityBase(
```

Comment:
This works if folks only want to use the default `OpportunityBase` class, but if they want to extend the `OpportunityBase` class with custom fields (like outlined in [this tutorial](https://commongrants.org/guides/using-python/#extend-the-sdks-opportunitybase-model)) this approach won't work.

An alternative would be to make this a generic class to which can pass a sub-class of `OpportunityBase` when they are implementing this ABC.